sk0-back/src/users/services/users.service.ts

```diff
@@ -53,4 +54,19 @@ export class UsersService {
     const user = await this.findOneById(id);
     return this.repo.remove(user);
   }
+
+  async validateUser(email: string, pass: string): Promise<User> {
+    const user = await this.findOneByEmail(email);
+    if (!user) {
+      throw new NotFoundException('User not found');
+    }
+
+    // Password decryption goes here in the future
+    const isPasswordMatching = pass === user.password;
+    if (!isPasswordMatching) {
+      throw new UnauthorizedException('Invalid credentials');
+    }
+
+    return user;
+  }
```

Comment:
I think it would be a better approach to create an auth.service.ts:
```typescript
import { Injectable } from '@nestjs/common';
import { LoginUserDto } from '../dtos/login-user.dto';
import { User } from '../user.entity';
import { UsersService } from './users.service';

@Injectable()
export class AuthService {
  constructor(private usersService: UsersService) {}
  async loginUser(loginUserDto: LoginUserDto): Promise<User> {
     const user = await this.findOneByEmail(email);
     if (!user) {
       throw new NotFoundException();
     }

     const isPasswordMatching = loginUserDto.password === user.password;
     if (!isPasswordMatching) {
        throw new UnauthorizedException();
     }
    
    return user;
  }
}

Reply:
I think I was drunk when I made this :D But yeah